jax/_src/scipy/stats/binom.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+
+import scipy.stats as osp_stats
+
+from jax import lax
+import jax.numpy as jnp
+from jax._src.numpy.util import _wraps, promote_args_inexact
+from jax._src.scipy.special import gammaln, xlogy, xlog1py
+from jax._src.typing import Array, ArrayLike
+
+
+@_wraps(osp_stats.nbinom.logpmf, update_doc=False)
+def logpmf(k: ArrayLike, n: ArrayLike, p: ArrayLike, loc: ArrayLike = 0) -> Array:
+    """JAX implementation of scipy.stats.binom.logpmf."""
+    k, n, p, loc = promote_args_inexact("binom.logpmf", k, n, p, loc)
+    y = lax.sub(k, loc)
+    comb_term = lax.sub(
+        gammaln(n + 1),
+        lax.add(gammaln(y + 1), gammaln(n - y + 1))
+    )
```

Comment:
This semantically correct, but numerically unfortunate for large-ish `n` and `y` close to either `n` or 0.  To wit, if `y` is close to 0, `gammaln(y + 1)` will be small, and this expression will produce a catastrophic cancellation between `gammaln(n + 1)` and `gammaln(n - y + 1)`.  Unfortunately, the solution is somewhat involved -- TFP had to reimplement the `lbeta` function in order to get this right: see [implementation](https://github.com/tensorflow/probability/blob/main/tensorflow_probability/python/math/special.py#L1854-L2129) and [usage](https://github.com/tensorflow/probability/blob/main/tensorflow_probability/python/distributions/binomial.py#L580-L582).  I also expect that code to be slower to compile and execute than what we have here.

I'm not sure of the right thing to do here.  We could
- Add a warning to the documentation that JAX prioritizes speed over corner-case accuracy, and this code will be somewhat inaccurate for large `n` and `y` close to `n` or 0; we can also direct users to TFP for the more accurate but presumably slower version.
- Fork TFP's implementation of `lbeta` and use it here.
- I assume we don't want to take TFP's `lbeta` as a dependency.